ground/src/main/java/com/google/android/ground/ui/datacollection/tasks/multiplechoice/MultipleChoiceAdapter.kt

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.android.ground.ui.datacollection.tasks.multiplechoice
+
+import android.view.LayoutInflater
+import android.view.View.GONE
+import android.view.View.VISIBLE
+import android.view.ViewGroup
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.ListAdapter
+import androidx.recyclerview.widget.RecyclerView
+import com.google.android.ground.R
+import com.google.android.ground.databinding.MultipleChoiceCheckboxItemBinding
+import com.google.android.ground.databinding.MultipleChoiceRadiobuttonItemBinding
+import com.google.android.ground.model.task.Option
+
+/**
+ * An implementation of [RecyclerView.Adapter] that associates [Option]s with the [ViewHolder]
+ * checkbox views.
+ */
+class MultipleChoiceAdapter(private val viewModel: MultipleChoiceTaskViewModel) :
+  ListAdapter<MultipleChoiceItem, RecyclerView.ViewHolder>(MultipleChoiceItemCallback()) {
+
+  override fun getItemViewType(position: Int): Int {
+    return when (getItem(position).type) {
+      MultipleChoiceItem.Type.CHECKBOX -> R.layout.multiple_choice_checkbox_item
+      else -> R.layout.multiple_choice_radiobutton_item
+    }
```

Comment:
Instead of using layout id as item type, can we add an integer value to the enum `MultipleChoiceItem.Type`?

Reply:
I opted for just passing in a boolean into the adapter which specifies its type.